drivers/flash/flash_mcux_flexspi.c

```diff
@@ -28,6 +28,17 @@ struct flash_flexspi_data {
 	size_t size[kFLEXSPI_PortCount];
 };
 
+bool flash_flexspi_is_running_xip(const struct device *dev)
+{
+#ifdef CONFIG_XIP
+	/* flexspi bus is labeled FLEXSPI & flexspi2 bus is labeled FLEXSPI1 */
+	return (strcmp(dev->name, "FLEXSPI") == 0 && IS_ENABLED(CONFIG_CODE_FLEXSPI))
+		|| (strcmp(dev->name, "FLEXSPI1") == 0 && IS_ENABLED(CONFIG_CODE_FLEXSPI2));
```

Comment:
@galak is there a macro in `include/devicetree.h` we could use to replace the strcmp here?

Reply:
Even for this runtime check? The i.MX series can have 2 `flexspi` busses active where one is running XIP and the other isn't.

Reply:
I do agree that the string compare isn't ideal especially because it always has to go through the whole name to check if there is a `1` at the end.
How about I compare the device's base address to what is predefined from the DTS?

Reply:
I did not succeed in creating a compile time `config` member because I couldn't get the macro magic to work, so I've added a `data` member.

If the compile time constant is still preferred, someone will need to give me pointers on how to do this :)

Reply:
Using `dev->name` for anything other than debugging is less and less of a good idea as time goes on.

If you have an enumeration of possible devices that `dev` might be and want to find out which one it is, why not store pointers to each struct device and just use pointer `dev == flexspi_dev`, `dev == flexspi2_dev` or so?

Reply:
@mbolivar-nordic I fully agree, and I've already moved the runtime check to the `init` function where I compare the device's base address [here](https://github.com/zephyrproject-rtos/zephyr/pull/33701/files#diff-76b6fe7650b1c121295403e474946fea492adc9b2565d2dd2f20ccef2b8478eaR124).

I think it could even be optimized to a compile time constant though.

Reply:
> @galak is there a macro in `include/devicetree.h` we could use to replace the strcmp here?

Forgot to mention: if you're comparing two known nodes, maybe [DT_SAME_NODE](https://docs.zephyrproject.org/latest/reference/devicetree/api.html#c.DT_SAME_NODE)?

Reply:
@mbolivar-nordic I tried your suggestion like so:

```C
#define MEMC_FLEXSPI_CFG_XIP(node_id) 						\
	COND_CODE_1(DT_SAME_NODE(node_id, DT_NODELABEL(flexspi)), 		\
		(IS_ENABLED(CONFIG_CODE_FLEXSPI)), 				\
		(COND_CODE_1(DT_SAME_NODE(node_id, DT_NODELABEL(flexspi2)), 	\
			(IS_ENABLED(CONFIG_CODE_FLEXSPI2)), (false))))
```

But the `DT_SAME_NODE` does not expand to a literal 1 or 0, wrapping it with `UTIL_BOOL` also didn't work.
This is as far as my macro magic knowledge goes atm :)

Reply:
Indeed it does not expand to literal 0 or 1. But neither does comparing the addresses. I will try to find time to take a more detailed look at the PR later but if you have a solution you're happy with that avoids strcmp, never mind me :)

Reply:
@mbolivar-nordic Your suggestion did get me to a solution if I turn the logic around!

```C
#define MEMC_FLEXSPI_CFG_XIP(node_id) 						\
	COND_CODE_1(CONFIG_CODE_FLEXSPI, 					\
		(DT_SAME_NODE(node_id, DT_NODELABEL(flexspi))), 		\
		(COND_CODE_1(CONFIG_CODE_FLEXSPI2, 				\
			(DT_SAME_NODE(node_id, DT_NODELABEL(flexspi2))), 	\
			(false))))
```

Thanks, will update